### PR TITLE
[DSV4][Triton] Support block sizes > 1 in paged MQA logits

### PR DESCRIPTION
**Pull request by skysnow2001**
## Motivation

When serving DeepSeek-V4-Flash with vLLM and AITER on 8 × `gfx1201` GPUs, decode uses AITER's paged MQA logits stage-1 Triton kernel (since the Gluon path is CDNA-only).

The stage-1 kernel assumed that each KV-cache block contained one token. vLLM uses a block size of 256 for this model, so the kernel incorrectly used each logical token position as an index into the block table. Once the logical token position exceeded the actual block-table width, it could read an invalid physical block ID and access memory outside the KV cache.

Because `--max-num-seqs` changes scheduler buffers, CUDA graph capture shapes, and the surrounding allocation layout, the invalid read did not fail identically for every value. Some configurations happened to read mapped memory, while others terminated workers with `HSA_STATUS_ERROR_MEMORY_APERTURE_VIOLATION`.

This PR generalizes the stage-1 path to support KV-cache block sizes greater than one.

## Technical details

The corrected indexing maps each logical token to its logical block and its offset within that block:

```text
logical_block_idx = logical_token_idx // block_size
block_offset      = logical_token_idx % block_size
physical_block    = block_table[batch, logical_block_idx]
```

FP8 values and FP32 scales are loaded using both the physical block index and the in-block token offset. The wrapper also splits the block-packed KV-cache storage at the actual value/scale boundary and passes the value/scale block and token strides into the Triton kernel.

## Test plan

I ran the same vLLM serving benchmarks before and after the fix on 8 × `gfx1201`, using 10 requests, concurrency 1, and two warm-up requests. I tested `--max-num-seqs` values `1`, `4`, `8`, `32`, and the vLLM default (`256`). For the default case, the `--max-num-seqs` option was omitted.

### Server command

```bash
VLLM_ROCM_USE_AITER=1 \
VLLM_ROCM_USE_AITER_RMSNORM=0 \
vllm serve deepseek-ai/DeepSeek-V4-Flash \
    --served-model-name deepseek-v4-flash \
    --host 0.0.0.0 \
    --port 8000 \
    --tensor-parallel-size 8 \
    --max-model-len 4096 \
    --gpu-memory-utilization 0.95 \
    --dtype auto \
    --kv-cache-dtype fp8 \
    --trust-remote-code \
    --compilation-config '{"cudagraph_mode": "FULL_AND_PIECEWISE"}'
```

For each non-default test, I appended one of:

```bash
--max-num-seqs 1
--max-num-seqs 4
--max-num-seqs 8
--max-num-seqs 32
```

### Benchmark command

```bash
vllm bench serve \
    --backend openai \
    --base-url http://localhost:8000 \
    --endpoint /v1/completions \
    --model deepseek-ai/DeepSeek-V4-Flash \
    --served-model-name deepseek-v4-flash \
    --dataset-name random \
    --random-input-len 1024 \
    --random-output-len 1024 \
    --num-prompts 10 \
    --max-concurrency 1 \
    --num-warmups 2 \
    --request-rate inf \
    --ignore-eos \
    --percentile-metrics 'ttft,tpot,itl,e2el'
```

The same command was repeated for input/output lengths `512/128`, `1024/1024`, and `2048/512`.

### Accuracy

GSM8K : 

```bash
lm_eval \
    --model local-completions \
    --tasks gsm8k \
    --model_args "model=deepseek-v4-flash,tokenizer=deepseek-ai/DeepSeek-V4-Flash,base_url=http://localhost:8000/v1/completions,num_concurrent=64,max_retries=3,tokenized_requests=False,trust_remote_code=True" \
    --batch_size auto
```

### Serving results

| Max sequences | Input → output tokens | Before: TTFT (ms) | Before: TPOT (ms) | After: TTFT (ms) | After: TPOT (ms) |
|---:|---:|---:|---:|---:|---:|
| 1 | 512 → 128 | **Fail** | **Fail** | 185.10 | 36.95 |
| 1 | 1024 → 1024 | **Fail** | **Fail** | 289.96 | 53.62 |
| 1 | 2048 → 512 | **Fail** | **Fail** | 474.03 | 38.87 |
| 4 | 512 → 128 | **Fail** | **Fail** | 187.96 | 37.07 |
| 4 | 1024 → 1024 | **Fail** | **Fail** | 239.34 | 38.02 |
| 4 | 2048 → 512 | **Fail** | **Fail** | 479.21 | 39.13 |
| 8 | 512 → 128 | 185.04 | 36.86 | 186.75 | 36.86 |
| 8 | 1024 → 1024 | 237.54 | 38.01 | 222.50 | 38.01 |
| 8 | 2048 → 512 | 485.67 | 39.68 | 488.76 | 38.90 |
| 32 | 512 → 128 | 185.07 | 37.01 | 185.85 | 36.91 |
| 32 | 1024 → 1024 | 226.37 | 38.13 | 227.67 | 37.89 |
| 32 | 2048 → 512 | 483.60 | 39.78 | 482.02 | 38.87 |
| 256 (default) | 512 → 128 | 138.80 | 35.20 | 138.17 | 35.11 |
| 256 (default) | 1024 → 1024 | 226.39 | 36.12 | 225.96 | 36.12 |
| 256 (default) | 2048 → 512 | 444.99 | 37.72 | 442.00 | 36.70 |

## GSM8K
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  | 0.95|±  | 0.006|
|     |       |strict-match    |     5|exact_match|↑  | 0.95|±  | 0.006|

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

**Comment by carlushuang** [comment]
@skysnow2001  please fix code format, this is hard requirement. Without it no other CI will trigger

**Comment by skysnow2001** [comment]
Fixed

**Review by cagrikymk — APPROVED** [review]
LGTM!

**Review by azaidy — APPROVED** [review]
LGTM!
